src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -260,5 +335,45 @@ public override BoundNode VisitInterpolatedString(BoundInterpolatedString node)
             }
             return result;
         }
+
+        [Conditional("DEBUG")]
+        private void AssertNoImplicitInterpolatedStringHandlerConversions(ImmutableArray<BoundExpression> arguments, bool allowConversionsWithNoContext = false)
+        {
+            if (allowConversionsWithNoContext)
+            {
+                foreach (var arg in arguments)
+                {
+                    if (arg is BoundConversion { Conversion: { Kind: ConversionKind.InterpolatedStringHandler }, ExplicitCastInCode: false, Operand: BoundInterpolatedString @string })
+                    {
+                        Debug.Assert(((BoundObjectCreationExpression)@string.InterpolationData!.Value.Construction).Arguments.All(
+                            a => a is BoundInterpolatedStringArgumentPlaceholder { ArgumentIndex: BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter }
+                                      or not BoundInterpolatedStringArgumentPlaceholder));
+                    }
+                }
+            }
+            else
+            {
+                Debug.Assert(arguments.All(arg => arg is not BoundConversion { Conversion: { IsInterpolatedStringHandler: true }, ExplicitCastInCode: false }));
+            }
+        }
+
+        private static int GetNextInterpolatedStringHandlerConversionIndex(ReadOnlySpan<BoundExpression> arguments)
+        {
+            for (int i = 0; i < arguments.Length; i++)
+            {
+                if (arguments[i] is BoundConversion { ConversionKind: ConversionKind.InterpolatedStringHandler, Operand: BoundInterpolatedString operand })
```

Comment:
>{ ConversionKind: ConversionKind.InterpolatedStringHandler, Operand: BoundInterpolatedString operand } [](http://example.com/codeflow?start=52&length=102)

Should we be checking ExplicitCastInCode here as well.

Reply:
This function will be gone in the next revision, but the code is inlined into the new `VisitArguments` function. I've asserted that `ExplicitCastInCode` is never true when the sub condition is true, rather than just checking here, because I want to have an assert verifying that we never hit such a condition and writing an assert in the other direction would be much more complex.